### PR TITLE
[twentymin] Fixed information extractor

### PR DESCRIPTION
**Pull request by goggle**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Since the site 20min.ch has changed to video player to embed its videos, it does not work anymore with youtube-dl. There was no bug report yet, but I have fixed that.

### CODE REVIEW

File: youtube_dl/extractor/twentymin.py [hunk 1]
```diff
@@ -77,8 +92,14 @@ def _real_extract(self, url):
...
 
         if not video_id:
+            params = self._html_search_regex(
+                r'<iframe[^>]+src="(?:https?:)?//www\.20min\.ch/videoplayer/videoplayer\.html\?params=(.+?[^"])"',
+                webpage, '20min embed URL', default=None)
             video_id = self._search_regex(
-                r'"file\d?"\s*,\s*\"(\d+)', webpage, 'video id')
+                r'.*videoId@(\d+)',
+                params, 'Video Id', default=None) if params is not None else ''
+            if not video_id:  # the article does not contain a video
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: You can't know whether it's an article with no video or video id pattern just changed. Original approach should be reverted.

File: youtube_dl/extractor/twentymin.py [hunk 2]
```diff
@@ -87,7 +108,7 @@ def _real_extract(self, url):
         return {
             'id': video_id,
             'display_id': display_id,
-            'url': 'http://speed.20min-tv.ch/%sm.flv' % video_id,
+            'url': 'http://podcast.20min-tv.ch/podcast/20min/%sh.mp4' % video_id,
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: SD quality is also available. Both should be extracted.

**Comment by goggle** [comment]
Alright, I've done the suggested changes. Is it ok now?

File: youtube_dl/extractor/twentymin.py [hunk 3]
```diff
@@ -77,8 +89,12 @@ def _real_extract(self, url):
                 r'^20 [Mm]inuten.*? -', '', self._og_search_title(webpage)), ' - News')
 
         if not video_id:
+            params = self._html_search_regex(
+                r'<iframe[^>]+src="(?:https?:)?//www\.20min\.ch/videoplayer/videoplayer\.html\?params=(.+?[^"])"',
+                webpage, '20min embed URL', default=None)
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: What's the point of `default=None`?

File: youtube_dl/extractor/twentymin.py [hunk 4]
```diff
@@ -77,8 +89,12 @@ def _real_extract(self, url):
...
             video_id = self._search_regex(
-                r'"file\d?"\s*,\s*\"(\d+)', webpage, 'video id')
+                r'.*videoId@(\d+)',
+                params, 'Video Id', default=None) if params is not None else ''
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: What's `default=None) if params is not None else ''` for?

**Comment by goggle** [comment]
This `if params is not None else ''` was a workaround for the case where `params == None`.

I think I've found a better solution now. Is that ok?

File: youtube_dl/extractor/twentymin.py [hunk 5]
```diff
@@ -77,8 +89,12 @@ def _real_extract(self, url):
                 r'^20 [Mm]inuten.*? -', '', self._og_search_title(webpage)), ' - News')
 
         if not video_id:
+            params = self._html_search_regex(
+                r'<iframe[^>]+src="(?:https?:)?//www\.20min\.ch/videoplayer/videoplayer\.html\?params=(.+?[^"])"',
+                webpage, '20min embed URL', default='')
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: What's the point of `default`?

File: youtube_dl/extractor/twentymin.py [hunk 6]
```diff
@@ -77,8 +89,12 @@ def _real_extract(self, url):
...
             video_id = self._search_regex(
-                r'"file\d?"\s*,\s*\"(\d+)', webpage, 'video id')
+                r'.*videoId@(\d+)',
+                params, 'Video Id', default='')
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: What's the point of `default`?

File: youtube_dl/extractor/twentymin.py [hunk 7]
```diff
@@ -87,8 +103,16 @@ def _real_extract(self, url):
...
             'description': description,
             'thumbnail': thumbnail,
+            'formats': [{
+                'format_id': 'sd',
+                'url': 'http://podcast.20min-tv.ch/podcast/20min/%s.mp4' % video_id,
+                'preference': -2
+            }, {
+                'format_id': 'hd',
+                'url': 'http://podcast.20min-tv.ch/podcast/20min/%sh.mp4' % video_id,
+                'preference': -1
```
Comment (dstftw) [inline youtube_dl/extractor/twentymin.py]: Code duplication.

**Comment by goggle** [comment]
Ok, so I removed `default=''` and removed the duplicated code.

Is that ok now?

**Pull request closed without merging**